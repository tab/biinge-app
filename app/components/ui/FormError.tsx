import React from "react"
import { View, Text } from "react-native"

import { formErrorStyles } from "styles"

type Props = {
  children: React.ReactNode
}

const FormErrorComponent = ({ children }: Props) => {
  return (
    <>
      {children && (
        <View style={formErrorStyles.container}>
          <Text style={formErrorStyles.text}>{children}</Text>
        </View>
      )}
    </>
  )
}

export default FormErrorComponent
