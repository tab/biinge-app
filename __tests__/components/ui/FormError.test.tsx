import React from "react"
import "react-native"
import { render } from "@testing-library/react-native"
import { it } from "@jest/globals"

import FormError from "components/ui/FormError"

describe("FormError", () => {
  const error = "Error message"

  it("renders correctly", () => {
    const tree = render(<FormError>{error}</FormError>)
    expect(tree).toMatchSnapshot()
  })
})
