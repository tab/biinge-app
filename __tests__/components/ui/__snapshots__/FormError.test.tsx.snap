// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormError renders correctly 1`] = `
<View
  style={
    {
      "marginBottom": 15,
    }
  }
>
  <Text
    style={
      {
        "color": "#9f0909",
        "fontSize": 15,
        "textAlign": "center",
      }
    }
  >
    Error message
  </Text>
</View>
`;
